### PR TITLE
Use map_element_get_direction in more places, create map_element_get_direction_with_offset()

### PR DESCRIPTION
**Pull request by Gymnasiast**
Needs a proper look, but should clean things up nicely.

### CODE REVIEW

**Comment by IntelOrca** [comment]
I think it would be better to have a general `direction_add` rather than `map_element_get_direction_with_offset`.

**Comment by Gymnasiast** [comment]
> I think it would be better to have a general direction_add rather than map_element_get_direction_with_offset.

This is shorter, though. I think it would be better to have both.

File: test/testpaint/compat.c [hunk 1]
```diff
@@ -129,6 +129,10 @@ int map_element_get_direction(const rct_map_element *element) {
     return element->type & MAP_ELEMENT_DIRECTION_MASK;
 }
 
+int map_element_get_direction_with_offset(const rct_map_element *element, int offset) {
+    return ((element->type & MAP_ELEMENT_DIRECTION_MASK) + offset) % 4;
```
Comment (IntelOrca) [inline test/testpaint/compat.c]: Now that offset is signed, can you make it `& 3`?

Comment (Gymnasiast) [inline test/testpaint/compat.c]: I opted for making the `offset` uint8 instead, since it failed to compile otherwise.

**Comment by duncanspumpkin** [comment]
I take it the compiler is clever enough to convert these `%4`'s into `&3` and `(+2) % 4` into `^2`.

File: src/openrct2/world/wall.cpp [hunk 2]
```diff
@@ -145,7 +145,7 @@ static bool WallCheckObstructionWithTrack(rct_scenery_entry * wall,
         return false;
     }
 
-    direction = (trackElement->type + direction) & 3;
+    direction = map_element_get_direction(trackElement);
```
Comment (duncanspumpkin) [inline src/openrct2/world/wall.cpp]: Somewhat causes #10196

**Pull request merged**